### ISSUE TITLE
Dice Loss resulting in unexpected logit outputs

### ISSUE THREAD

**Issue by trchudley** [qubvel-org/segmentation_models.pytorch#900]
Hi, thanks for such a great package!

I'm currently training a simply U-net on a binary image classification problem (in this case, identifying water in satellite imagery). I am exploring different zsegmentation_models_pytorch` loss functions to train the model. I have been using a loss function initially to build and test the model:

```python
loss_function = smp.losses.FocalLoss(mode="binary", alpha=alpha, gamma=gamma"
```

After training the model, this model produces a relatively sensible output in both raw logit values and as a probability value after a sigmoid activation layer is applied:

![focalloss](https://github.com/user-attachments/assets/1627bb49-43af-4308-82c6-cabf51a57715)

(NB: this is just a test run, the final accuracy doesn't really matter at this point)

I have been looking to explore and switch to dice loss. My understanding is that, using the `from_logits` variable, I could simply drop-and-replace the loss function with the `FocalLoss` class as follows:

```python
loss_function = smp.losses.DiceLoss(mode="binary", from_logits=True")
```

This new model, applied to the same image, results in the following:

![diceloss](https://github.com/user-attachments/assets/8ead64b4-3d9b-4437-9cb7-6edf1589c275)

Looking at the logit output, this is great - the new dice-loss-trained model appears to qualitatively perform even better than the focal-loss-trained model! However, the raw outputs are not scaled around zero any more. Instead, raw outputs are all positive, scaled between approximately ~400 and ~9000 (depending on what image the model is being applied). As a result, applying a sigmoid activation does not create a nice probability distribution between zero and one - instead, the apparent probabilities are all now 1, due to the all-positive logit distribution.

I've examined the source code and I can't see anything that would result in such a difference. Am I missing something that results in this not being a drag-and-drop replacement to create probabilistic model predictions?

**Comment by qubvel** [comment]
Hi @trchudley, it should be a drop-in replacement. If `from_logits=True` with binary mode is specified sigmoid function is used inside the loss function. It's an interesting observation, you can see that for [example](https://github.com/qubvel-org/segmentation_models.pytorch/blob/main/examples/binary_segmentation_intro.ipynb) notebook for binary segmentation DiceLoss function works fine and predicted logits are distributed around 0.
